core/src/main/java/com/linecorp/armeria/server/HttpServerHandler.java

```diff
@@ -353,7 +362,13 @@ private void handleRequest(ChannelHandlerContext ctx, DecodedHttpRequest req) th
             HttpResponse serviceResponse;
             try {
                 req.init(reqCtx);
-                serviceResponse = service.serve(reqCtx, req);
+                serviceResponse = HttpResponse.from(() -> {
```

Comment:
If `serviceEventLoop == eventLoop`, we can call `service.serve` right away without delegating to the `serviceEventLoop`.

Reply:
- `reqCtx.push()` above must be executed in the synchronous boundary of `service.serve(reqCtx, req)`.
- How about extract the `try (SafeCloseable ignored = reqCtx.push()) { ... }` block as a separate method and it execute it in a `serviceEventLoop`.
  ```java
  private void handleRequest(...) {

      final boolean needsDirectExecution;
      if (serviceEventLoop == config.workerGroup()) {
          needsDirectExecution = true;
      }

      if (needsDirectExecution) {
          serveRequest(...);
      } else {
          serviceEventLoop.execute(() -> serveRequest(...));
      }
  }  

  private void serveRequest(...) {
      try (SafeCloseable ignored = reqCtx.push()) {
           ...
      }
  }
  ```